docs/interact-with-the-stack.md

```diff
@@ -0,0 +1,163 @@
+
+To rebuild and recompose a service first run:
+
+```bash
+make build-<name of module>
+```
+
+> [!NOTE]
+> Containers that have an associated volume will need to have the volume rebuild with `make docker-volumes` before being able to successfully compose the container.
+
+Then compose the container with:
+
+```bash
+make compose-<name of module>
+```
+
+> [!IMPORTANT]
+> Some services can't be rebuilt individually without causing issues with the stack, if you are facing issues with modules related to auth, it is recommended to rebuild the entire stack to ensure everything is in sync.
+
+## Non-destructive Rebuild
+
+To rebuild the CanDIGv2 without destroying data in postgres or keycloak the make target `rebuild-keep-data` with:
+
+```bash
+make rebuild-keep-data
+```
+
+> [!NOTE]
+> If there are changes that have changed the structure of the database or impacted the versions of other CANDIG_DATA_MODULES this way of rebuilding cannot be used.
+
+## Destructive Cleanup 
+
+Use the following steps to clean up running CanDIGv2 services in a docker-compose configuration. 
+
+> [!CAUTION] 
+> Note that these steps are destructive and will remove **ALL** logs, containers, secrets, volumes, networks, certs, and images. If you are using docker in a shared environment (i.e. with other non-CanDIGv2 containers running) please consider running the cleanup steps manually instead.
+
+The following steps are performed by `make clean-all`:
+
+```bash
+# 1. delete log files
+make clean-logs
+
+# 2. stop and remove running stacks
+make clean-compose
+
+# 3. stop and remove remaining containers
+make clean-containers
+
+# 4. remove all configs/secrets from docker and local dir
+make clean-secrets
+
+# 5. remove all docker volumes and local data dir
+make clean-volumes
+
+# 6. delete all cached images
+make clean-images
+```
+
+See the [Makefile](../Makefile) for the exact commands that each of these targets runs.
+
+## Rebuild entire stack from scratch
+
+1. Perform any backups of data necessary if in a non-testing environment. (see [backup and restore doc](backup-restore-candig.md) for detailed instructions.)
+
+2. Clean up the current containers with `make clean-all`
+
+3. When complete, build all containers again with `make build-all`
+
+
+## Troubleshooting
+
+### Conda env not activated
+
+If you get an error when running a make command, something like:
+
+```
+bash: python: command not found
+```
+or an error message about `dotenv` not being found.
+
+Ensure the candig conda environment is activated in your terminal with `conda activate candig`.
+
+### docker volumes not remade
+
+If you get an error where after cleaning an individual service, when composing, it gets stuck at 
+
+```
+waiting for x service to start ...
+```
+
+Use CTRL + c to exit the process then try running `make docker-volumes` and then try composing again with `make compose-<name of service>`
```

Comment:
I keep meaning to add a bit to make compose that adds back any known external volumes...